src/BotwinExtensions.cs

```diff
@@ -40,15 +42,38 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
                     handler = CreateModuleBeforeAfterHandler(module, route);
 
                     var finalHandler = CreateFinalHandler(handler, statusCodeHandlers);
-                    
+
                     routeBuilder.MapVerb(route.verb, route.path, finalHandler);
                 }
             }
 
             return builder.UseRouter(routeBuilder.Build());
         }
 
-        
+        private static void Add405Handler(IApplicationBuilder builder, IEnumerable<BotwinModule> srvs)
+        {
+            var systemRoutes = new List<(string verb, string route)>();
```

Comment:
`Dictionary<path, IEnumerable<verb>` would probably read nicer in this method, and might be quicker than all of the linq too

Reply:
to load the dictionary you'd have to do ` systemRoutes.Add(route.path, module.Routes.Where(x=>x.path == route.path).Select(x=>x.verb));`

i think this whole piece of code could be improved though

Reply:
Hmm, you're not wrong. There's gotta be a nicer way of getting that code in there

Reply:
Wouldn't something like this be more efficient in the long run?

```
var systemRoutes = srvs.SelectMany(m => m.Routes)
    .GroupBy(r => "/" + r.path)
    .ToDictionary(
        g => g.Key, 
        g => new HashSet<string>(
            g.Select(r => r.verb).Distinct()
        )
    ); // IDictionary<string, HashSet<string>>
```

We than could just do:
```
if (systemRoutes.ContainsKey(strippedPath) &&
    !systemRoutes[strippedPath].Contains(context.Request.Method)) // almost O(1) access time
{
    context.Response.StatusCode = 405;
    return;
}
```

(the LINQ part may be simplified into a `foreach` if it appears too cryptical)